Use default keymap for codemirror accept completion including 'Enter'

fisher-alice opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->
This PR updates the CodeMirror autocompletion in `lab2`'s `CodeEditor` so that the default keymap is used. This includes 'Enter' for accepting completion: https://codemirror.net/docs/ref/#autocomplete.completionKeymap

https://github.com/code-dot-org/code-dot-org/pull/61799 updated autocomplete to use 'Tab' instead of 'Enter'. We want to continue to allowing 'Tab' to accept completion. However, the other default key mappings are desirable so I updated config to add 'Tab', but then use the default key bindings since they match what we currently have implemented.

I tested in both `pythonlab` and `weblab2` and both 'Enter' and 'Tab' keys accept completion. The up/down arrows continue to be used to navigate through completion option.

'Esc' is supposed to close the completion dropdown, but currently the key is mapped to moving user out of the code editor - ~~not sure if this is expected.~~ cc: @hannahbergam. 
UPDATE: Hannah confirmed this is undesirable. Created a follow-up ticket.

## After update

Using keyboard navigation to access autocompletion dropdown, navigation, and selection - 'Enter' for selection - in `pythonlab` and `weblab2`.

https://github.com/user-attachments/assets/7ea94f7f-1e36-425c-885b-68a5be4ef501


## Links

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

- Jira: https://codedotorg.atlassian.net/browse/CT-1219

## Testing story
Tested locally on `pythonlab` and `weblab2` levels.

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work
'Esc' is supposed to close the completion dropdown, but currently when 'Esc' key is pressed, the user is moved out of the code editor: https://codedotorg.atlassian.net/browse/SL-1241

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

hannahbergam reviewed (commented):
I'm surprised CodeMirror doesn't stopPropagation with something like this so that any other listeners don't get triggered. I'm then curious about what we were replacing the default keymap with and whether we can just override the keys to use the default keys but also stopPropagation (or preventDefault- whichever works)

fisher-alice commented:
> I'm surprised CodeMirror doesn't stopPropagation with something like this so that any other listeners don't get triggered. I'm then curious about what we were replacing the default keymap with and whether we can just override the keys to use the default keys but also stopPropagation (or preventDefault- whichever works)

I'll go ahead and log a separate ticket for the 'Escape' key as this is the current behavior on `prod`. Thank you!

hannahbergam reviewed (commented):
Can you help me understand the 'before' behavior?

fisher-alice commented:
> Can you help me understand the 'before' behavior?

Thank you for this question! The 'before' behavior is that our autocomplete keybindings are almost the same other than replacing 'Enter' with 'Tab' for accepting completion. 

https://github.com/code-dot-org/code-dot-org/blob/3da722ab6b6114c3469f5a7ebc673297d331e3c6/apps/src/lab2/views/components/editor/editorConfig.ts#L33-L44

I added more info in the PR description, and also updated the PR so that `autocompleteKeybindings` now just adds the 'Tab' keybinding since we can use the default keybindings for other keys.

hannahbergam reviewed (approved):
Works for me! Thank you and nice cleanup!
